src/pynxtools_xps/vms/vamas_comment_handler.py

```diff
@@ -0,0 +1,170 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# pylint: disable=too-many-instance-attributes
+
+from typing import List, Literal, Any, Dict, Tuple
+
+from pynxtools_xps.vms.casa_data_model import CasaProcess
+from pynxtools_xps.phi.spe_pro_phi import PhiParser
+from pynxtools_xps.kratos.metadata_kratos import KratosParser
+
+from pynxtools_xps.reader_utils import (
+    convert_pascal_to_snake,
+)
+
+
+def handle_comments(
+    comment_list: List[str], comment_type: Literal["header", "block"]
+) -> Dict[str, Any]:
+    comments = {}
+
+    special_keys: Tuple[str, ...] = (
+        "Casa Info Follows",
+        "SpecsLab Prodigy",
+        "SOFH",
+        "Creation",
+    )
+
+    for keyword in special_keys:
+        # if any(keyword in line for line in comment_list):
+        indices = [i for i, line in enumerate(comment_list) if keyword in line]
+
+        if indices:
+            index = indices[0]
+            if keyword == "Casa Info Follows":
+                if comment_type == "header":
+                    special_comments = [comment_list[index]]
+                if comment_type == "block":
+                    special_comments = comment_list[index:]
+                    # TODO: only extract casa comments
+                casa_comments, no_of_casa_lines = _handle_casa_comments(
+                    special_comments, comment_type
+                )
+                comments.update(casa_comments)
+                del comment_list[index:no_of_casa_lines]
+
+            if keyword == "SpecsLab Prodigy":
+                special_comments = [comment_list[index]]
+                comment_list = comment_list[index + 1 :]
+                comments.update(_handle_prodigy_header_comments(special_comments))
+
+            if keyword == "SOFH":
+                end_index = [
+                    i for i, line in enumerate(comment_list) if "EOFH" in line
+                ][0]
+                special_comments = comment_list[index : end_index + 1]  # type: ignore[assignment]
+                del comment_list[index : end_index + 1]
+                comments.update(_handle_phi_header_comments(special_comments))
+
+            if keyword == "Creation":
+                end_index = [
+                    i for i, line in enumerate(comment_list) if "X-ray Power" in line
+                ][0]
+                special_comments = comment_list[index : end_index + 1]  # type: ignore[assignment]
+                del comment_list[index : end_index + 1]
+                comments.update(_handle_kratos_block_comments(special_comments))
```

Comment:
Sometimes it's good to disentangle the code by introducing functions with a dict to call the separate parts. I think it's still fine here but if this grows in the future you might want to change it for better readability

Reply:
I agree. For now, I'll keep it, but if there is more to come in the future, I'll distentangle the function.